src/rez/rex.py

```diff
@@ -11,6 +13,7 @@
 from contextlib import contextmanager
 from string import Formatter
 from collections.abc import MutableMapping
+from typing import Iterable
```

Comment:
You could support all python versions by using `rez.utils.typing` like you did for protocol.

Reply:
`typing.Iterable` exists for all supported versions of python (since python 3.5, I believe).

Reply:
The thing is that it's deprecated in 3.9 in favor of collections.abc.Iterable.